_posts/2019-01-10-future-workloads.md

```diff
@@ -0,0 +1,503 @@
+---
+layout: post
+title: keywords: future workloads scheduling
+author: jguitana
+---
+
+One of the most relevant Scala primitives for concurrency is the `Future`. It represents a value which is either complete or failed, and is scheduled to run outside the program main flow, in some other thread.
+
+Future enables us to tackle the following concurrency problems:
+1. Let other components execute when one is waiting on IO.
+2. Make small programs communicate to build larger ones.
+3. Use all the CPU's in our machines.
+4. Enable human-computer interaction (GUIs, terminal).
+
+In this post we will look at three distinct kinds of workloads we can run in a Future: `cpu bound`, `blocking` and `asynchronous`, and gain some intuition about how they perform.
+
+These workloads generally fit in the problems described above as follows:
+1. blocking, async
+2. blocking, async
```

Comment:
What's the difference between number 1 and 2?

Reply:
The numbers  ( ͡° ͜ʖ ͡°)

Reply:
Yeah I guess that has to be more explicit

Reply:
u.u. I'm also not sure about which they should be. Probably non-blocking, async is one of them?

Reply:
I just mentioned them without introducing them. It comes after lol.